addons/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/IntRSModule.java

```diff
@@ -0,0 +1,128 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.satel.internal.protocol;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.TooManyListenersException;
+
+import javax.naming.ConfigurationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Represents Satel INT-RS module. Implements methods required to connect and
+ * communicate with that module over serial protocol.
+ *
+ * @author Krzysztof Goworek - Initial contribution
+ * @since 1.7.0
+ */
+public class IntRSModule extends SatelModule {
+
+    private final Logger logger = LoggerFactory.getLogger(IntRSModule.class);
+
+    private String port;
+
+    /**
+     * Creates new instance with port and timeout set to specified values.
+     *
+     * @param port
+     *            serial port the module is connected to
+     * @param timeout
+     *            timeout value in milliseconds for connect/read/write
+     *            operations
+     * @throws ConfigurationException
+     *             unconditionally throws this exception as it is not
+     *             implemented yet
+     */
+    public IntRSModule(String port, int timeout) {
+        super(timeout);
+
+        this.port = port;
+    }
+
+    @Override
+    protected CommunicationChannel connect() {
+        logger.info("Connecting to INT-RS module at {}", this.port);
+
+        try {
+            CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(this.port);
+            SerialPort serialPort = portIdentifier.open("org.openhab.binding.satel", 2000);
+            serialPort.setSerialPortParams(19200, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            serialPort.enableReceiveTimeout(this.getTimeout());
+            // RXTX serial port library causes high CPU load
+            // Start event listener, which will just sleep and slow down event
+            // loop
+            serialPort.addEventListener(new SerialPortEventListener() {
+                @Override
+                public void serialEvent(SerialPortEvent ev) {
+                    try {
+                        logger.trace("RXTX library CPU load workaround, sleep forever");
+                        Thread.sleep(Long.MAX_VALUE);
+                    } catch (InterruptedException e) {
+                    }
+                }
+            });
+            serialPort.notifyOnDataAvailable(true);
+
+            logger.info("INT-RS module connected successfuly");
+            return new SerialCommunicationChannel(serialPort);
+        } catch (NoSuchPortException e) {
+            logger.error("Port {} does not exist", this.port);
```

Comment:
See above about error logging, here I would like to add that the message which log here should go to the detailMessage of the status update.

Reply:
Changed the code a bit and now those errors are propagated to the bridge and sent as "detailMessage" when updating bridge status.